kani-compiler/src/kani_middle/provide.rs

```diff
@@ -61,6 +63,12 @@ fn run_kani_mir_passes<'tcx>(
     tracing::debug!(?def_id, "Run Kani transformation passes");
     let mut transformed_body = stubbing::transform(tcx, def_id, body);
     stubbing::transform_foreign_functions(tcx, &mut transformed_body);
+    if KaniAttributes::for_item(tcx, def_id).is_contract_generated()
+        && (stubbing::get_stub_key(tcx, def_id).is_some()
+            || KaniAttributes::for_item(tcx, def_id).has_recursion())
```

Comment:
If we apply the transformation in all items that are contract generated, we will ended up instantiating `any_modifies` for the replace function every time, even if we are only checking the contract, because the function is always included during contract instrumentation. Thus, we ended up with the same problem as before. The key part here is to only apply the transformation if we are using `stub_verified` or in the presence of recursion. I added some comments to make this clear.